src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -162,9 +162,11 @@ public String getScript() {
 
         private int pid;
         private final long startTime = System.currentTimeMillis();
+        private EnvVars envVars = new EnvVars();
```

Comment:
This initializer is useless—it is always overridden by the constructor.

You can set fields in restored objects using a `readResolve` method, but it would be simpler to just mark this field `@CheckForNull` as a reminder that it will occasionally be unset, and have `exitStatus` behave accordingly.